components/heirloomSettingsModal/index.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from "react";
+
+import styles from "./modal.module.css";
+import LoadingSpinner from "../loadingSpinner";
+import { useMemorial } from "../../lib/memorial";
+import SelectedRectangle from "../selectedRectangle";
+
+export default function HeirloomSettingsModal({ open, onClose, memorialID }) {
+  // this prevents the background from scrolling when modal is open
+  useEffect(() => {
+    document.body.style.overflow = open ? "hidden" : "auto";
+    return () => {
+      document.body.style.overflow = "auto";
+    };
+  }, [open]);
+
+  // this prevents clicks from inside modal from closing it
+  function stopPropagation(e) {
+    e.stopPropagation();
+  }
+
+  const { loading, memorial } = useMemorial(memorialID);
```

Comment:
Again, great work with this use function. Very similar to the useUser hook but this controls information of the memorial so to keep the code modular.